backup/README.md

```diff
@@ -2,7 +2,98 @@
 
 As the database has grown (from a few 10s of MB up to GB) the dump-and-restore approach we are using is becoming painful.  Downloading from S3 takes quite a while (especially verification with duplicati) and the restore (`pg_restore`) takes a long time to rebuild the indices.
 
-## Barman
+## Deployment
+
+Barman is running on `annex.montagu.dide.ic.ac.uk` as container `montagu-barman`
+
+```
+git clone https://github.com/vimc/montagu-db
+cd montagu-db/backup
+pip3 install -r requirements.txt
+```
+
+Then interact with the barman container the `./barman-montagu` command:
+
+```
+$ ./montagu-barman --help
+Set up barman (Postgres streaming backup) for montagu
+
+Usage:
+  barman-montagu setup [options] <host>
+  barman-montagu status
+  barman-montagu barman [--] [<args>...]
+  barman-montagu recover [--wipe-first] [<backup-id>]
+  barman-montagu wipe-recover
+  barman-montagu destroy
+
+Options:
+  --configure-cron          Configure a once-a-minute cron job
+  --password-group=<group>  Password group [default: production]
+  --image-tag=<tag>         Barman image tag [default: master]
+  --pull-image              Pull image before running?
+  --no-clean-on-error       Don't clean up container/volumes on error
+```
+
+These commands create and pass through to the `barman` process in a long running container that is called `barman-montagu`
+
+To set up barman:
+
+```
+./barman-montagu setup --image-tag=i1559 --pull localhost
+```
+
+(replace `localhost` with the name or IP address of the host running the montagu postgres container on port 5432).
+
+To install the cron job pass `--configure-cron` (which will also require running with `sudo`).  Or to set up cron on an already running barman:
+
+```
+sudo ./barman-montagu setup --configure-cron localhost
+```
+
+To see a set of status information run
+
+```
+./barman-montagu status
+```
+
+To interact with barman directly, either do `docker exec barman-montagu barman ...` or use
+
+```
+
+./barman-montagu barman -- --help
+```
+
+(the `--` is often optional but disambiguates arguments to `barman-montagu` and for barman).  For example listing files in the most recent backup might look like:
+
+```
+./barman-montagu barman list-files montagu latest
+```
+
+or getting information about the latest backup:
+
+```
+./barman-montagu barman show-backup montagu latest
+```
+
+(see [the `barman` docs](http://docs.pgbarman.org/release/2.0) for more commands).
+
+To dump out the latest copy of the database to recover from:
+
+```
+./barman-montagu recover --wipe-first
```

Comment:
When would you want to not wipe first?

Reply:
I'm not sure here - it _appears_ that restoring onto an existing path will probably work and I'd like to see if that's the case (or reserve the ability to).  It will make things heaps faster if we can because it will avoid copying around many GB of data